scripts/dep_update.sh

```diff
@@ -27,43 +43,47 @@ if [[ "${READ_ONLY_TEST:-true}" == "true" ]]; then
   exit 0
 fi
 
-if [[ -z "$(git status -s)" ]]
-then
-  echo "Repo is clean, prep script made no changes to be committed."
-else
-  echo "Repo is dirty, preparing to propose changes."
-  # split URL into array separated by '/'s
-  IFS='/ ' read -r -a url_split <<< "${REPO_URL}"
-  # owner/org is whatever is between the second and third / in the url
-  owner="${url_split[3]}"
-  # repo is anything after owner/
-  repo=${REPO_URL#https://github.com/${owner}/}
-  ssh_url="git@github.com:${owner}/${repo}"
-  message="This change is the result of running gating/update_dependencies/*
-  See update_dependencies in https://rpc-openstack.atlassian.net/wiki/spaces/RE/pages/19005457/RE+for+Projects for more details."
+# split URL into array separated by '/'s
+IFS='/ ' read -r -a url_split <<< "${REPO_URL}"
+# owner/org is whatever is between the second and third / in the url
+owner="${url_split[3]}"
+# repo is anything after owner/
+repo=${REPO_URL#https://github.com/${owner}/}
+ssh_url="git@github.com:${owner}/${repo}"
 
-  # Get / Create Jira issue
+if [[ -n "$(git status -s)" ]] || [[ ${start_sha} != $(git rev-parse --verify HEAD) ]]; then
   echo "Looking for Jira issue, will create if not found."
+  issue_message="This issue was generated automatically by the Jenkins job ${RE_JOB_NAME}.
+  See update_dependencies in https://rpc-openstack.atlassian.net/wiki/spaces/RE/pages/19005457/RE+for+Projects for more details."
   jira_summary="Update ${repo}:${BRANCH} dependencies"
   issue=$(python ${WORKSPACE}/rpc-gating/scripts/jirautils.py \
         --user "${JIRA_USER}" \
         --password "${JIRA_PASS}" \
         get_or_create_issue \
           --project "${JIRA_PROJECT_KEY}" \
           --summary "${jira_summary}" \
-          --description "${message}" \
+          --description "${issue_message}" \
           --label RE_DEP_UPDATE \
           --label jenkins \
           --label "${repo}_${branch}"
   )
   echo "Issue: ${issue}"
+fi
 
-  echo "Committing changes"
-  pr_branch="${issue}_${BRANCH}_dep_update"
-  title="${issue} Update ${BRANCH} dependencies"
+if [[ -z "$(git status -s)" ]]
+then
+  echo "Repo is clean, prep script made no changes to be committed."
+else
+  echo "Repo is dirty, committing changes."
+  title="${issue} Update ${BRANCH} third-party dependencies"
+  message="This change is the result of running gating/update_dependencies/*
+  See update_dependencies in https://rpc-openstack.atlassian.net/wiki/spaces/RE/pages/19005457/RE+for+Projects for more details."
   git commit -a -m "${title}" -m "${message}"
+fi
 
+if [[ ${start_sha} != $(git rev-parse --verify HEAD) ]]; then
```

Comment:
Why a separate if block here?  If we commit in the if block above, then this will always be true, right?

Reply:
The component dependency update creates its own commit separately (lines 28-36). The separation is so that a pull request is made regardless of where the change comes from.